src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -808,6 +808,14 @@ private SecurityStatusPal GenerateToken(ReadOnlySpan<byte> inputBuffer, ref byte
                         }
                     }
 
+#if TARGET_ANDROID
```

Comment:
Can we move it to `AcceptSecurityContext` and `InitSecurityContext`? This is were context is created on other platforms. And the `_securityContext` is `ref`.

Reply:
The problem I'm trying to solve with this piece of code is that I need to pass the reference to the SslStream to the context. I haven't found a better way of doing it while keeping the signature of `AcceptSecurityContext` and `InitSecurityContext` unchanged. You previously suggested passing the SslStream proxy through `SafeFreeCredentials` but I wasn't able to pull that off. I'll try to improve this admittedly ugly and confusing initialization.

Reply:
So I moved the `SslStreamProxy` into `SslAuthenticationOptions` on Android. The code is definitely cleaner this way.